### PR TITLE
Decisions decisions: Add decision-making flow for stale threads

### PR DESCRIPTION
**Pull request by Shadowfiend**
contributes to THESIS-58

### Notes
This adds a decision making process to threads with call-to-action buttons running different interactions such as tagging a user. Will archive a channel after 24 hours of inactivity or if the user decides to archive.

Screenshot 2023-08-19 at 00 05 02

### CODE REVIEW

**Comment by zuuring** [comment]
contributes to THESIS-58

**Comment by zuuring** [comment]
@Shadowfiend Changed the `requestFollowUpAction()` around a bit here to resolve the issue with interaction errors happening when you tagged yourself. Otherwise, this feels at a good place to start testing on a specific channel (TBD).

File: discord-scripts/thread-management/check-thread-archiving.ts
Comment (Shadowfiend) [inline discord-scripts/thread-management/check-thread-archiving.ts:260]: Ohp, why'd we comment these out?

**Comment by Shadowfiend** [comment]
Just realized I can't approve this PR because I was the author. @zuuring you're good to approve it and then merge when ready on Monday.

File: discord-scripts/thread-management/check-thread-archiving.ts
Comment (zuuring) [reply discord-scripts/thread-management/check-thread-archiving.ts:261]: Brought back `isInRecreationalCategory` but left `!isInPermittedCategoryOrChannel` since that check is causing some issues against testing channels. We can implement that as the main call once we remove `!isInTestingChannel(thread)`

**Review by zuuring — APPROVED** [review]
Let's merge this one! w000t

**Pull request merged**
